### PR TITLE
APP-500: Offer footer

### PR DESCRIPTION
**Pull request by hugokallstrom**
### Checklist

- [ ] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt [hunk 1]
```diff
@@ -396,6 +401,25 @@ class OfferAdapter(
...
+                removeDiscount: () -> Unit,
+                marketManager: MarketManager
+            ) {
+                if (data is OfferModel.Footer) {
+                    binding.chatButton.setHapticClickListener {
+                        it.context.startActivity(ChatActivity.newInstance(it.context, true))
+                    }
+                    val link = itemView.context.getString(R.string.OFFER_FOOTER_GDPR_INFO, data.url)
+                    binding.text.createWebLink(link)
+                }
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt:420]: I'm a fan of the following pattern:

```suggestion
                if (data !is OfferModel.Footer) {
                	return invalid(data)
                }
                binding.chatButton.setHapticClickListener {
                    it.context.startActivity(ChatActivity.newInstance(it.context, true))
                }
                val link = itemView.context.getString(R.string.OFFER_FOOTER_GDPR_INFO, data.url)
                binding.text.createWebLink(link)
```

For the simple reason that it creates one less indentation 🙂

File: app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt [hunk 2]
```diff
@@ -396,6 +401,25 @@ class OfferAdapter(
...
+                fragmentManager: FragmentManager,
+                tracker: OfferTracker,
+                removeDiscount: () -> Unit,
+                marketManager: MarketManager
+            ) {
+                if (data is OfferModel.Footer) {
+                    binding.chatButton.setHapticClickListener {
+                        it.context.startActivity(ChatActivity.newInstance(it.context, true))
+                    }
+                    val link = itemView.context.getString(R.string.OFFER_FOOTER_GDPR_INFO, data.url)
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt:418]: Does this text key contain `HTML` now? 🤔

File: app/src/main/res/layout/offer_footer.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        android:text="@string/OFFER_FOOTER_BUTTON_TEXT" />
+
+    <TextView
+        android:id="@+id/text"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/base_margin_sextuple"
+        android:clickable="true"
+        android:focusable="true"
+        android:linksClickable="true"
```
Comment (hedvigoscar) [inline app/src/main/res/layout/offer_footer.xml:33]: Are all of the following attributes required in order to support in-view links? 🤔

**Review by hedvigoscar — COMMENTED** [review]
Neat!

File: app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt (hunk 2 above)
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt:418]: Yes unfortunately. Is there a better way of handling this?

File: app/src/main/res/layout/offer_footer.xml (hunk 3 above)
Comment (hugokallstrom) [reply app/src/main/res/layout/offer_footer.xml:33]: Yes :O

**Review by hedvigoscar — APPROVED** [review]

**Pull request merged**